lib/hmmlearn/hmm.py

```diff
@@ -955,11 +940,41 @@ def _accumulate_sufficient_statistics(self, stats, X, lattice,
 
         with np.errstate(under="ignore"):
             post_comp_mix = post_comp[:, :, None] * post_mix
-        stats['post_comp_mix'].append(post_comp_mix)
 
         stats['post_mix_sum'] += post_comp_mix.sum(axis=0)
         stats['post_sum'] += post_comp.sum(axis=0)
 
+        # means stats
+        stats['m_n'] += np.einsum('ijk,il->jkl', post_comp_mix, X)
+
+        if 'c' in self.params:  # covariance stats
+            centered = X[:, None, None, :] - self.means_
+
+            def outer_f(x):  # Outer product over features.
+                return x[..., :, None] * x[..., None, :]
+
+            def norm_last(x):  # Much faster than (x**2).sum(-1).
+                return np.einsum('...i,...i', x, x)
+
+            if self.covariance_type == 'full':
+                centered_dots = outer_f(centered)
+                c_n = np.einsum('ijk,ijklm->jklm', post_comp_mix,
+                                     centered_dots)
+            elif self.covariance_type == 'diag':
+                centered2 = np.square(centered, out=centered)  # reuse
+                c_n = np.einsum('ijk,ijkl->jkl', post_comp_mix, centered2)
+            elif self.covariance_type == 'spherical':
+                centered_norm = norm_last(centered)
+                c_n = np.einsum('ijk,ijk->jk', post_comp_mix, centered_norm)
+            elif self.covariance_type == 'tied':
+                centered_dots = outer_f(centered)
+                c_n = np.einsum('ijk,ijklm->jlm', post_comp_mix, centered_dots)
+
+            if 'c_n' in stats:
+                stats['c_n'] += c_n
+            else:
+                stats['c_n'] = c_n
```

Comment:
I guess you could init c_n to zeros in initialize_sufficient_statistics to avoid that check.